### ISSUE TITLE
Http/2 support

### ISSUE THREAD

**Issue by andreatosato** [ThreeMammals/Ocelot#1124]
Do you have plan to supporto http/2 for REST Api?
No for gRPC, only https and http/2 rest api.
Thanks a lot

**Comment by andreatosato** [comment]
https://github.com/ThreeMammals/Ocelot/blob/65710f4a94c8175669e5f5818a7ac002bc48db27/src/Ocelot/Requester/HttpClientBuilder.cs#L61-L64

HttpClient must declare:

`Version = new Version(2, 0)` in HttpRequestMessage
or
`DefaultRequestVersion = new Version(2, 0)`

**Comment by TomPallister** [comment]
Is this a breaking change?

**Comment by andreatosato** [comment]
No, I think that user must configure HTTP version.
In json (better way, each API must configure correct HTTP) or via Startup file (generic config for all API).

**Comment by TomPallister** [comment]
OK cool! I guess the user would setup if the Ocelot webserver is handling HTTP2 connections so this is just for the downstream request?

**Comment by andreatosato** [comment]
Yes, only for downstream request

**Comment by TomPallister** [comment]
This should be simple enough to do, add an option in the config probably per ReRoute and globally. Then pass this down to where the HttpClient is instantiated. Ocelot creates a HttpClient per ReRoute would this need to change?

**Comment by andreatosato** [comment]
I think that option per ReRoute fit better for this case.
Globally options is not very useful.

My particular case:
I host in Kestrel API and gRPC service.
For gRPC I need Kestrel HTTP2 so, my API must use HTTP2.

**Comment by TomPallister** [comment]
OK cool, if I have time I will give it a go.

If you need this feature quickly maybe try and submit a PR with the required changed!

**Comment by andreatosato** [comment]
I must implement this feature for internal workshop.
This week I'm very busy but I can try next week.

**Comment by andreatosato** [comment]
Quickly fix:
https://github.com/ThreeMammals/Ocelot/blob/fc3b6fdb8b7ee00e16e8f9e3d73992de98fab191/src/Ocelot/Request/Mapper/RequestMapper.cs#L23-L28

```cs
 var requestMessage = new HttpRequestMessage()
                {
                    Content = await MapContent(request),
                    Method = MapMethod(request, downstreamReRoute),
                    RequestUri = MapUri(request),
                    Version = new Version(2, 0)
                };
```

**Comment by raman-m** [comment]
Closed by #1127
